bazel/openssl_repositories.bzl

```diff
@@ -0,0 +1,58 @@
+load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")
```

Comment:
Is it possible to let grpc to use the same ssl library as jaeger exporter?If we have a program which links both otlp grpc exporter and jaeger exporter, there may be symbol conflicts of libressl(used by grpc by default) and openssl.

Reply:
`gRPC` is using `boringssl`. There is a [trick](https://groups.google.com/g/grpc-io/c/nBjdFT-d1Bo/m/QQ1VuOhCAQAJ) to make gRPC use openssl. Though we are already using the boringssl that comes form gRPC in our [CI](https://github.com/open-telemetry/opentelemetry-cpp/blob/f380fcba45112b91deb53d14daae785cdbf320b5/ci/setup_grpc.sh#L43), while using `openssl` for Jaeger exporter.

Reply:
boringssl and openssl have some symbols with same name. I just wonder whether they have 100% ABI compatibility with each other. If so, it's fine to use one implementation of some functions and global variables and use the rest functions and global variables of another.

Reply:
I believe they are not [compatible](https://boringssl.googlesource.com/boringssl/+/HEAD/PORTING.md).
I would suggest investigating this in another issue/PR. We should have the same issue in CMake.

Reply:
In cmake, we can use `-DgRPC_SSL_PROVIDER=package` and `-DOPENSSL_ROOT_DIR=<OpenSSL installation directory>` when building grpc and grpc will use openssl to replace boringssl. But I'm not familiar with bazel and not know how to do this in bazel.

Reply:
yes that should be the fix for `CMake`. For `bazel` we should define a new library with the name `boringssl` before importing `gRPC`.
I can propose the fix in another PR for both `CMake` and `bazel`?

Reply:
Sounds good and thanks.